app/src/main/java/com/example/erabook/fragments/discoverGoogle/DiscoverDetailsFragment.kt

```diff
@@ -85,14 +86,28 @@ class DiscoverDetailsFragment : Fragment() {
                     requireContext()
                 )
             }
+            when (bookItem?.accessInfo?.viewability) {
+                "ALL_PAGES" -> {
+                    if (bookItem.accessInfo.pdf?.isAvailable == true) {
+                        buyBook.visibility = View.GONE
+                        readBook.visibility = View.VISIBLE
+                        readBook.setOnClickListener {
+                            openLinkBookDownload(
+                                bookItem.accessInfo.pdf.downloadLink,
```

Comment:
You are passing nullable values to the extension function. With this implementation you are not handling null values.